### PR TITLE
WT-13381 Improve startup logs

### PR DESCRIPTION
**Pull request by kommiharibabu**
Add additional logs in the flow of WiredTiger recovery and also convert the existing logs into INFO logs to log them every time when WiredTiger is performing recovery.

### CODE REVIEW

File: src/txn/txn_recover.c [hunk 1]
```diff
@@ -1152,8 +1159,10 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
             eviction_started = true;
         }
 
-        __wt_verbose_multi(session,
-          WT_DECL_VERBOSE_MULTI_CATEGORY(((WT_VERBOSE_CATEGORY[]){WT_VERB_RECOVERY, WT_VERB_RTS})),
+        __wt_verbose_level_multi(session,
+          WT_DECL_VERBOSE_MULTI_CATEGORY(
+            ((WT_VERBOSE_CATEGORY[]){WT_VERB_RECOVERY, WT_VERB_RECOVERY_PROGRESS, WT_VERB_RTS})),
```
Comment (etienneptl) [inline src/txn/txn_recover.c:1164]: Should `WT_VERB_RECOVERY_ALL` include `WT_VERB_RTS` so we can simply use `WT_VERB_RECOVERY_ALL`?

Comment (kommiharibabu) [reply src/txn/txn_recover.c:1164]: Hmm, No. RTS has a specific log checker that verifies all the log messages that are logged with WT_VERB_RTS. If we include recovery messages also as part of the WT_VERB_RTS, we need to add log handling for them also.

File: test/suite/test_verbose04.py [hunk 2]
```diff
@@ -118,9 +118,9 @@ def test_verbose_all(self):
             session.compact(uri)
             session.close()
 
-        # At this time, no verbose messages should be generated with the following set of operations and the verbosity level
-        # WT_VERBOSE_INFO (0), hence we don't expect any output.
-        with self.expect_verbose(['all:0'], self.all_verbose_categories, self.is_json, False) as conn:
+        # At this time, only INFO verbose messages should be generated with the following set of
+        # operations and the verbosity level WT_VERBOSE_INFO (0).
+        with self.expect_verbose(['all:0'], self.all_verbose_categories, self.is_json) as conn:
```
Comment (korteland) [inline test/suite/test_verbose04.py:123]: This code change is correct, but it seems like there's a philosophical change here? We've never logged anything at INFO before, although I can see there are now some recent exceptions for compact.

The documentation says INFO is for "describing statuses and actions performed by WiredTiger" so it doesn't seem inappropriate, just a break from tradition.

File: src/txn/txn_recover.c [hunk 3]
```diff
@@ -813,6 +813,9 @@ __recovery_file_scan_prefix(WT_RECOVERY *r, const char *prefix, const char *igno
 static int
 __recovery_file_scan(WT_RECOVERY *r)
 {
+    __wt_verbose_level_multi(r->session, WT_VERB_RECOVERY_ALL, WT_VERBOSE_INFO, "%s",
```
Comment (korteland) [inline src/txn/txn_recover.c:816]: There's a lot of emphasis on recovery with these changes, but the BF linked on the ticket indicated recovery wasn't the problem. What about adding some more in `wiredtiger_open`?

File: test/suite/test_verbose04.py (hunk 2 above)
Comment (kommiharibabu) [reply test/suite/test_verbose04.py:123]: Yes, in the KTLO stream a couple of people raised an opinion on a couple of WT logs to be printed with a default configuration. This is the step towards printing all the WT INFO messages into the MongoDB logs. We plan to add/convert a couple of existing verbose messages into the INFO messages.

Comment (korteland) [reply test/suite/test_verbose04.py:123]: Cool, that makes sense. Thanks for explaining things.

File: src/txn/txn_recover.c (hunk 3 above)
Comment (kommiharibabu) [reply src/txn/txn_recover.c:816]: I added a couple more logs in the flow of `wiredtiger_open`.

File: src/conn/conn_open.c [hunk 4]
```diff
@@ -275,5 +277,8 @@ __wti_connection_workers(WT_SESSION_IMPL *session, const char *cfg[])
     /* Start the checkpoint cleanup thread. */
     WT_RET(__wt_checkpoint_cleanup_create(session, cfg));
 
+    __wt_verbose_info(
+      session, WT_VERB_RECOVERY, "%s", "WiredTiger utility threads are started successfully");
```
Comment (raviprakashgiri29) [inline src/conn/conn_open.c:281]: Can you please rephrase it to "WiredTiger utility threads started successfully." ?

**Comment by raviprakashgiri29** [comment]
@kommiharibabu Do you plan to add logging to the beginning of every component as part of this ticket, or should we address it in a separate ticket?

**Comment by kommiharibabu** [comment]
@raviprakashgiri29 Adding logs for every component init would be too much. I combined a couple of functions where they can take some time to process, in those locations I added the logs. This ticket only handles the startup flow.

Individual module log levels needs to adjusted/add new logs as part of a separate ticket.

File: src/conn/conn_open.c (hunk 4 above)
Comment (kommiharibabu) [reply src/conn/conn_open.c]: Done.

**Review by raviprakashgiri29 — APPROVED** [review]
LGTM

**Review by raviprakashgiri29 — APPROVED** [review]

File: src/txn/txn_recover.c (hunk 3 above)
Comment (korteland) [reply src/txn/txn_recover.c:816]: Thanks - that's the kind of thing I had in mind when I made the ticket.

**Review by korteland — APPROVED** [review]
LGTM!

File: src/conn/conn_api.c [hunk 5]
```diff
@@ -3159,6 +3159,7 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler, const char *c
     conn->page_size = __wt_get_vm_pagesize();
 
     /* Now that we know if verbose is configured, output the version. */
+    __wt_verbose_info(session, WT_VERB_RECOVERY, "%s", "opening the WiredTiger library");
     __wt_verbose(session, WT_VERB_VERSION, "%s", WIREDTIGER_VERSION_STRING);
```
Comment (etienneptl) [inline src/conn/conn_api.c:3163]: Would that make sense to combine the two?
```suggestion
    __wt_verbose_info(session, WT_VERB_RECOVERY, "opening the WiredTiger library %s", WIREDTIGER_VERSION_STRING);
```

**Review by etienneptl — APPROVED** [review]
One suggestion, LGTM

File: src/conn/conn_api.c (hunk 5 above)
Comment (kommiharibabu) [reply src/conn/conn_api.c:3163]: I thought of doing it, but the WiredTiger version always points to the version number and it has a very old date. I feel it can cause confusion to TS, so I dropped it.

Comment (etienneptl) [reply src/conn/conn_api.c:3163]: Fair enough - thank you for checking.
